src/convert/convertContext/decomposedCustomLabelFinalizer.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { join } from 'node:path';
+import { ensure, JsonMap } from '@salesforce/ts-types';
+import type { CustomLabel } from '@jsforce/jsforce-node/lib/api/metadata';
+import { customLabelHasFullName } from '../../utils/metadata';
+import { MetadataType } from '../../registry';
+import { XML_NS_KEY, XML_NS_URL } from '../../common/constants';
+import { JsToXml } from '../streams';
+import { WriterFormat } from '../types';
+import { ConvertTransactionFinalizer } from './transactionFinalizer';
+
+type CustomLabelState = {
+  /*
+   * Incoming child xml (CustomLabel) keyed by label fullname
+   */
+  customLabelByFullName: Map<string, CustomLabel>;
+};
+
+/**
+ * Merges child components that share the same parent in the conversion pipeline
+ * into a single file.
+ *
+ * Inserts unclaimed child components into the parent that belongs to the default package
+ */
+export class DecomposedCustomLabelsFinalizer extends ConvertTransactionFinalizer<CustomLabelState> {
```

Comment:
The class name doesn't match the file name.

Reply:
nice catch, I'll fix that